### PR TITLE
fix: skip spec.nodeName env populated for downwardAPI

### LINKED ISSUE #509: dynamic pod env can't pass-through to Leaf cluster
**What happened**:

When Kafka statefulset resources are orchestrated across clusters using Kosmos, they utilize dynamic environment variables, such as obtaining the nodeName of the current pod. However, in the current version of Kosmos, when pods are dispatched to leaf clusters, they are assigned virtual node names instead of the actual leaf cluster node names. This discrepancy leads to operational issues with the pods.

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 3
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      nodeName: kosmos-kafka
      tolerations:
        - key: kosmos.io/node
          operator: Equal
          value: "true"
      containers:
        - name: nginx
          image: nginx:latest
          ports:
            - containerPort: 80
          env:
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.name
            - name: POD_NAMESPACE
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.namespace
            - name: JMX_PORT
              value: "9999"
            - name: NODEIP
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: status.hostIP
            - name: NODENAME
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: spec.nodeName
            - name: PODIP
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: status.podIP
            - name: IS_KAFKA_FROZEN
              value: "false"
            - name: KAFKA_HEAP_OPTS
              value: "-Xms8192 -Xmx8192M"
            - name: LOG_DIRS
              value: "/kafka/data-0"
```

**What you expected to happen**:
the pod should get the real env 

**How to reproduce it (as minimally and precisely as possible)**:

image

**Anything else we need to know?**:

**Environment**:
- Kosmos version:
- Others:

### PR DESCRIPTION
**Pull request by village-way**
#### What type of PR is this?

/kind bug
#### Which issue(s) does this PR fix?

Fixes #
#### Special notes for your reviewer:

Fixes #509 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/utils/lifted/env.go [hunk 1]
```diff
@@ -479,8 +479,9 @@ func podFieldSelectorRuntimeValue(fs *corev1.ObjectFieldSelector, pod *corev1.Po
 		return "", err
 	}
 	switch internalFieldPath {
-	case "spec.nodeName":
-		return pod.Spec.NodeName, nil
+	// skip spec.nodeName populated by vk, for kafka broker scenario
```
Comment (duanmengkk) [inline pkg/utils/lifted/env.go]:
```suggestion
	// skip spec.nodeName 
```

**Comment by duanmengkk** [comment]
cherry-pick to release-0.3.0

File: pkg/utils/lifted/env.go (hunk 1 above)
Comment (village-way) [reply pkg/utils/lifted/env.go]: fixed

**Pull request merged**